jobs/build/gen-assembly/Jenkinsfile

```diff
@@ -92,55 +99,68 @@ node {
             if (!params.IN_FLIGHT_PREV) {
                 error('IN_FLIGHT_PREV is required. If there is no in-flight release, use "none".')
             }
-            doozerParams = "--group openshift-${params.BUILD_VERSION} --data-path ${params.DOOZER_DATA_PATH}"
-            if (params.LIMIT_ARCHES.trim()) {
-                if (!params.CUSTOM) error('LIMIT_ARCHES can only be used with custom assemblies.')
-                doozerParams += " --arches ${commonlib.cleanCommaList(params.LIMIT_ARCHES)}"
+            if (!params.ASSEMBLY_NAME) {
+                error('ASSEMBLY_NAME is required.')
             }
         }
 
-        stage("get-nightlies") {
-            def nightly_args = ""
-            if (params.ALLOW_PENDING) nightly_args += "--allow-pending"
-            if (params.ALLOW_REJECTED) nightly_args += " --allow-rejected"
-            if (params.ALLOW_INCONSISTENCY) nightly_args += " --allow-inconsistency"
-            for (nightly in commonlib.parseList(params.NIGHTLIES)) {
-                nightly_args += " --matching ${nightly.trim()}"
-            }
-            cmd = "${doozerParams} get-nightlies ${nightly_args}"
-            nightlies = commonlib.parseList(buildlib.doozer(cmd, [capture: true]))
-        }
         stage("gen-assembly") {
-            withEnv(['KUBECONFIG=/home/jenkins/kubeconfigs/art-publish.app.ci.kubeconfig']) {
-                nightly_args = ""
-                for (nightly in nightlies) {
-                    nightly_args += " --nightly ${nightly.trim()}"
+            buildlib.cleanWorkdir("./artcd_working")
+            sh "mkdir -p ./artcd_working"
+            def cmd = [
+                "artcd",
+                "-v",
+                "--working-dir=./artcd_working",
+                "--config", "./config/artcd.toml",
+            ]
+
+            if (params.DRY_RUN) {
+                cmd << "--dry-run"
+            }
+            cmd += [
+                "gen-assembly",
+                "--ocp-build-data-url", params.DOOZER_DATA_PATH,
```

Comment:
For consistency within our tooling, I suggest naming this param as `data-path`, as seen for Elliott and Doozer

Reply:
Sounds good to me.